12.js

```diff
@@ -0,0 +1,31 @@
+const stringMeAlong = string => {
+  let longestSubstring = ''
+  let substring
+
+  for (let index = 0; index < string.length; index++) {
+    substring = findLongestSubstringStartingAtIndex(string, index)
```

Comment:
instead of passing the string and the startIndex you could just pass the subset of the string

```js
substring = findLongestSubstringStartingAtIndex(string.slice(index))
```
then you index is always 0 within `findLongestSubstringStartingAtIndex`